image/config.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package image
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/opencontainers/image-spec/schema"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+)
+
+type cfg struct {
+	User         string
+	Memory       int64
+	MemorySwap   int64
+	CPUShares    int64 `json:"CpuShares"`
+	ExposedPorts map[string]struct{}
```

Comment:
You're currently not translating this to anything (because the runtime spec [doesn't have knobs for Linux networking][1]).  @philips, can you explain how you expect [higher-levels notice and apply this configuration][2]?

[1]: https://github.com/opencontainers/image-spec/issues/87#issuecomment-221463049
[2]: https://github.com/opencontainers/image-spec/issues/87#issuecomment-221623046